portal-kernel/src/com/liferay/portal/kernel/util/SystemProperties.java

```diff
@@ -220,4 +225,68 @@ public static void set(String key, String value) {
 		load(classLoader);
 	}
 
+	private static class ExtendedProperties extends Properties {
```

Comment:
@W2jx The logic here is not easy to understand. Let's do this:
1. Read everything from the file and only keep the line which do not lead with "#"
2. Then pass in the extracted result to jdk Properties to load the values